tests/test_process_data.py

```
@@ -0,0 +1,97 @@
+
+    The result is verified to be a DataFrame and not empty.
+    """
+    # Mock ConnectorFactory to return mock_csv_connector for both source and target
+    with patch.object(ConnectorFactory, 'get_connector', return_value=mock_csv_connector):
+        test_case = {
+            'test': {
+                'source': {'type': 'csv', 'location': '/path/to/source.csv'},
+                'target': {'type': 'csv', 'location': '/path/to/target.csv'}
+            }
+        }
+        source_df, target_df = process_data(test_case)
```

Ruff (0.8.2)

`process_data` may be undefined, or defined from star imports

(F405)